packages/dds/tree/src/simple-tree/treeApi.ts

```diff
@@ -208,6 +209,36 @@ export const treeNodeApi: TreeNodeApi = {
 	},
 };
 
+/**
+ * Returns a schema for a value if the value is a {@link TreeNode} or a {@link TreeLeafValue}.
+ * Returns undefined for other values.
+ */
+function tryGetSchema<T>(value: T): undefined | TreeNodeSchema<string, NodeKind, unknown, T> {
+	type TOut = TreeNodeSchema<string, NodeKind, unknown, T>;
+	switch (typeof value) {
+		case "string":
+			return stringSchema as TOut;
+		case "number":
+			return numberSchema as TOut;
+		case "boolean":
+			return numberSchema as TOut;
+		case "object": {
+			if (isTreeNode(value)) {
+				// This case could be optimized, for example by placing the simple schema in a symbol on tree nodes.
+				return tryGetSimpleNodeSchema(getFlexNode(value).schema) as TOut;
+			}
+			if (value === null) {
+				return nullSchema as TOut;
+			}
+			if (isFluidHandle(value)) {
+				return handleSchema as TOut;
+			}
+		}
+		default:
+			return undefined;
```

Comment:
We want to support `"null": nullSchema` too I imagine. How did the `null` unit test pass without it?

Reply:
`typeof  null` is "object" not "null", and its handled on like 231.

Reply:
Oh right - there's no `typeof "null"` :)